src/commands/lambda/functions/instrument.ts

```diff
@@ -134,17 +147,58 @@ export const calculateUpdateRequest = async (
   }
   let needsUpdate = false
 
-  // Update Handler
-  const expectedHandler = HANDLER_LOCATION[runtime]
-  if (config.Handler !== expectedHandler) {
-    needsUpdate = true
-    updateRequest.Handler = HANDLER_LOCATION[runtime]
+  // If a layer version is set for Java or a custom runtime.
+  // We need to inform the customer to only set the extension argument.
+  if (RUNTIME_LOOKUP[runtime] === RuntimeType.JAVA || RUNTIME_LOOKUP[runtime] === RuntimeType.CUSTOM) {
+    if (settings.layerVersion !== undefined) {
+      throw new Error(
+        `Only the extensionVersion argument should be set for the ${runtime} runtime. Please remove the layerVersion argument from the instrument command.`
```

Comment:
According to the README, the arguments are `--extension-version` and `--layer-version` (i.e. kebab-case, not camelCase)